homeassistant/components/climate/ecobee.py

```diff
@@ -22,16 +22,23 @@
 _LOGGER = logging.getLogger(__name__)
 
 ATTR_FAN_MIN_ON_TIME = 'fan_min_on_time'
+ATTR_RESUME_ALL = 'resume_all'
 
 DEPENDENCIES = ['ecobee']
 
 SERVICE_SET_FAN_MIN_ON_TIME = 'ecobee_set_fan_min_on_time'
+SERVICE_RESUME_PROGRAM = 'ecobee_resume_program'
 
 SET_FAN_MIN_ON_TIME_SCHEMA = vol.Schema({
     vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
     vol.Required(ATTR_FAN_MIN_ON_TIME): vol.Coerce(int),
 })
 
+RESUME_PROGRAM_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Optional(ATTR_RESUME_ALL): cv.boolean,
```

Comment:
This should be required by two reasons. First, without any argument the lookup in the service function will return None, and you will pass a string version of that to the API. I don't think that is good. You could add a default, but see my second reason. Second, to have scene support working the user must specify the argument as a state attribute in the scene config. I think it's more obvious to the user how the scene support works for ecobee if we require the attribute also when calling the service.